# Installation

### Environment & Dependencies

- [Cell Ranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger) V. 3.0.2

This pipeline analyses use  R studio V. 4.0.2 and the packages: 

- [SoupX](https://github.com/constantAmateur/SoupX) V.1.4.5
- DoubletFinder V.2.0.3
- Seurat V. 3.1.2
- Monocle3 V.
- gProfier2 V.0.1.9

Docker is provided (link to Docker)

# Data

Raw files are deposited in ArrayExpress, under accession number E-MTAB-9467



# Data Mapping 

CellRanger V. 3.0.2 and human reference GRCh38-1.2.0 were used in these datasets

Running command as following 

cellranger count --id= sample id \
                              --transcriptome=PATH_TO_REF \
                              --fastqs=PATH_TO_RAW_FILES \
                              --sample=sample name \
                              --chemistry=SC3Pv2 \
                              --expect-cells=5000



# Workflow of downstream analyses

1. Quality control 
   - Correct the expression profiles using SoupX
   - Exclude the cells that have high percentage of mitochondrial counts
   - Exclude doublet using DoubletFinder  

2. Data integration

   - Integrate all the samples using SCTransform normalization generated by Seurat

3. Highly variable genes and pathway analyses

   - Investigate Pearson Correlation Coefficient and PCAs
   - Construct unsupervised hierarchical clustering from top 500 genes on PCs
   - Investigate GO enrichments of each cluster using gProfier2 

4. Trajectory and pseudotime analyses 

   - Create Monocle's object using the information from Seurat's object 

   - Construct trajectory and pseudotime using Monocle3
   - Set Naive at the root of pseudotime





















